core/CMakeLists.txt

```diff
@@ -50,3 +52,35 @@ foreach(LIB ${PRIVATE_LIBS})
 endforeach()
 
 target_include_directories(knoting PUBLIC ${CMAKE_SOURCE_DIR}/core/inc)
+
+# BGFX
+# TODO: Move these to a function which can be called from whichever runtime
+
+foreach(SHADER_FILE ${KNOTING_SHADERS})
+    get_filename_component(SHADER_NAME ${SHADER_FILE} NAME)
+    get_filename_component(PARENT_DIR ${SHADER_FILE} DIRECTORY)
+    string(REGEX REPLACE "^${CMAKE_SOURCE_DIR}/res/shaders" "" PARENT_DIR ${PARENT_DIR})
+
+    set(SHADER_TYPE "")
+    if("${SHADER_NAME}" MATCHES "vs_*")
+        message(STATUS "Found vertex shader ${SHADER_NAME}")
+        set(SHADER_TYPE VERTEX)
+    elseif("${SHADER_NAME}" MATCHES "fs_*")
+        message(STATUS "Found fragment shader ${SHADER_NAME}")
+        set(SHADER_TYPE FRAGMENT)
+    elseif("${SHADER_NAME}" MATCHES "cs_*")
+        message(STATUS "Found compute shader ${SHADER_NAME}")
+        set(SHADER_TYPE COMPUTE)
+    else()
+        message(STATUS "Found & Skipped unknown shader ${SHADER_NAME}")
+        continue()
+    endif()
```

Comment:
Are we expecting to use other shader types later and adding them or expecting to only use these three?